compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -220,6 +220,7 @@ pub struct ResolverAstLowering {
 
     /// Information about functions signatures for delegation items expansion
     pub delegation_fn_sigs: LocalDefIdMap<DelegationFnSig>,
+    pub delegation_resolution_info: LocalDefIdMap<DelegationResolutionInfo>,
```

Comment:
```suggestion
    pub delegation_sig_resolution_ids: LocalDefIdMap<ast::NodeId>,
```
The wrapper struct is not yet necessary, IMO.
Also, this needs a comment explaining what these ids are.